### PR TITLE
`azurerm_eventhub_namespace` - adding wait before deletion

### PR DESCRIPTION
**Pull request by xiaxyi**
fix failed acctest:
```
Error: deleting Namespace (Subscription: "*******"
Resource Group Name: "xxxx"
Namespace Name: "xxx"): namespaces.NamespacesClient#Delete: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="Conflict" Message="Namespace provisioning in transition. For more information visit https://aka.ms/eventhubsarmexceptions. CorrelationId: e82de239-409f-8f6f-e732-49925f9fc597"
--- FAIL: TestAccEventHubNamespace_withAliasConnectionString (659.94s)
```

### CODE REVIEW

File: internal/services/eventhub/eventhub_namespace_resource.go [hunk 1]
```diff
@@ -619,6 +619,11 @@ func resourceEventHubNamespaceDelete(d *pluginsdk.ResourceData, meta interface{}
 		return err
 	}
 
+	// need to wait for namespace status to be ready before deleting.
+	if err := waitForEventHubNamespaceStatusToBeReady(ctx, meta, *id, d.Timeout(pluginsdk.TimeoutUpdate)); err != nil {
+		return fmt.Errorf("waiting for eventHub namespace %s state to be ready error: %+v", *id, err)
+	}
```
Comment (tombuildsstuff) [inline internal/services/eventhub/eventhub_namespace_resource.go:625]: this indicates that another resource isn't marking itself as Created/Updated correctly, which resource is causing this?

Comment (xiaxyi) [reply internal/services/eventhub/eventhub_namespace_resource.go:625]: its disasterRecoveryConfig resource. The situation is the as the same as servicebus namespace. The fix was made in pr:https://github.com/hashicorp/terraform-provider-azurerm/pull/18706

Comment (tombuildsstuff) [reply internal/services/eventhub/eventhub_namespace_resource.go:625]: So is this fixed in `main`?

Comment (tombuildsstuff) [reply internal/services/eventhub/eventhub_namespace_resource.go:625]: Got it - so we should look to revert #18706 and then fix the actual issue here - this type of fix ends up causing larger issues elsewhere unfortunately.

**Comment by katbyte** [comment]
@xiaxyi - just wondering if this is still in progress?

**Comment by xiaxyi** [comment]
@katbyte Thanks for the comment, yes, this is still in the progress, any questions or concerns to the code change?

**Comment by katbyte** [comment]
As per toms comment, we should be solving this on the resource that is causing the issue, ie the DRC. the wait shouldn't be in the delete, it should be in the update/create//the resource that is putting the eventhub into this state. not the delete of the eventhub. the delete should be erroring if the resource is "undeletable" - this doesn't fix the case if tf creates it and then something else wants to work on it

**Comment by xiaxyi** [comment]
Thanks @katbyte for the comments, sure, I will check to see if we are able to get the DCR's provisioning state.

I would still like to explain that the wait function used in the deletFunc is to get the eventhub namespace status not the sub-resource ie DCR's status, thus, the eventhub namespace deletion may not be affected if DCR is not existed.

File: internal/services/eventhub/eventhub_namespace_resource.go (hunk 1 above)
Comment (xiaxyi) [reply internal/services/eventhub/eventhub_namespace_resource.go:625]: Thanks @tombuildsstuff for the comment. I left my comment in the thread, hope that explains your concern. :)

**Review by katbyte — APPROVED** [review]
Thanks for the clarification, LGTM 🦀

**Pull request merged**